src/go/pt-galera-log-explainer/translate/translate.go

```diff
@@ -0,0 +1,206 @@
+package translate
+
+import (
+	"encoding/json"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/percona/percona-toolkit/src/go/pt-galera-log-explainer/utils"
+)
+
+type translationUnit struct {
+	Value     string
+	Timestamp time.Time
+}
+
+type translationsDB struct {
+	// 1 hash: only 1 IP. wsrep_node_address is not dynamic
+	// if there's a restart, the hash will change as well anyway
+	HashToIP map[string]translationUnit
+
+	// wsrep_node_name is dynamic
+	HashToNodeNames map[string][]translationUnit
+	IPToNodeNames   map[string][]translationUnit
+
+	// incase methods changed in the middle, tls=>ssl
+	IPToMethods map[string][]translationUnit
+	rwlock      *sync.RWMutex
+}
+
+var AssumeIPStable bool = true
+
+var db = translationsDB{}
+
+func init() {
+	initTranslationsDB()
+}
+
+func initTranslationsDB() {
+	db = translationsDB{
+		HashToIP:        map[string]translationUnit{},
+		HashToNodeNames: map[string][]translationUnit{},
+		IPToMethods:     map[string][]translationUnit{},
+		IPToNodeNames:   map[string][]translationUnit{},
+		rwlock:          &sync.RWMutex{},
```

Comment:
If to use mutex by value, then we don't need to initialize it at all.
```suggestion
```

Reply:
Thanks, I was not sure